src/include/migraphx/op/reshape.hpp

```diff
@@ -76,22 +76,24 @@ struct reshape
         const bool has_negative_dim_attr = neg_dim_num < dims.size();
         // construct output dynamic shape from dims attribute
         std::vector<shape::dynamic_dimension> output_dyn_dims(dims.size());
-        std::transform(dims.begin(),
-                       dims.end(),
-                       input_dyn_dims.begin(),
```

Comment:
this loop would run for `dims.size()` times.  I do not see difference in logic between current change and new logic.

Reply:
Simply that we don't access `input_dyn_dims` out of bounds